packages/fluentui/react-northstar/src/components/Avatar/AvatarLabel.tsx

```diff
@@ -0,0 +1,191 @@
+  iconPosition?: 'start' | 'end';
+
+  /** A AvatarLabel can contain an image. */
+  image?: ShorthandValue<ImageProps>;
+
+  /** A AvatarLabel can position its image at the start or end of the layout. */
+  imagePosition?: 'start' | 'end';
+}
+
+export type AvatarLabelStylesProps = Pick<AvatarLabelProps, 'circular' | 'color' | 'imagePosition' | 'iconPosition'> & {
+  hasImage: boolean;
+  hasIcon: boolean;
+  hasActionableIcon: boolean;
+};
+export const avatarlabelClassName = labelClassName;
+
+/**
+ * A AvatarLabel allows user to classify content.
+ */
+export const AvatarLabel: ComponentWithAs<'span', AvatarLabelProps> & FluentComponentStaticProps = props => {
+  const context = useFluentContext();
+  const { setStart, setEnd } = useTelemetry(AvatarLabel.displayName, context.telemetry);
+  setStart();
+
+  const {
+    accessibility,
+    children,
+    className,
+    circular,
+    color,
+    content,
+    icon,
+    iconPosition,
+    design,
+    styles,
+    variables,
+    image,
+    imagePosition,
+  } = props;
+
+  const getA11Props = useAccessibility(accessibility, {
+    debugName: AvatarLabel.displayName,
+    rtl: context.rtl,
+  });
+  const { classes, styles: resolvedStyles } = useStyles<AvatarLabelStylesProps>(AvatarLabel.displayName, {
+    className: avatarlabelClassName,
+    mapPropsToStyles: () => ({
+      hasActionableIcon: _.has(icon, 'onClick'),
+      hasImage: !!image,
+      hasIcon: !!icon,
+      circular,
+      color,
+      imagePosition,
+      iconPosition,
+    }),
+    mapPropsToInlineStyles: () => ({ className, design, styles, variables }),
+    rtl: context.rtl,
+  });
+
+  const ElementType = getElementType(props);
+  const unhandledProps = useUnhandledProps(AvatarLabel.handledProps, props);
+
+  if (childrenExist(children)) {
+    const element = (
+      <ElementType
+        {...getA11Props('root', {
+          className: classes.root,
+          ...rtlTextContainer.getAttributes({ forElements: [children] }),
+          ...unhandledProps,
+        })}
+      >
+        {children}
+      </ElementType>
+    );
+    setEnd();
+
+    return element;
+  }
+
+  const imageElement = Image.create(image, {
+    defaultProps: () => ({
+      styles: resolvedStyles.image,
+    }),
+  });
+  const iconElement = Box.create(icon, {
+    defaultProps: () => ({
+      styles: resolvedStyles.icon,
+    }),
+  });
+  const contentElement = Box.create(content, {
+    defaultProps: () => ({
+      styles: resolvedStyles.content,
+    }),
+  });
+
+  const startImage = imagePosition === 'start' && imageElement;
+  const startIcon = iconPosition === 'start' && iconElement;
+  const endIcon = iconPosition === 'end' && iconElement;
+  const endImage = imagePosition === 'end' && imageElement;
+
+  const element = (
+    <ElementType
+      {...getA11Props('root', {
+        className: classes.root,
+        ...unhandledProps,
+      })}
+    >
+      {startImage}
```

Comment:
do we need all of these slots? They don't work properly in the current Avatar anyway: https://codesandbox.io/s/vqhr2?module=/example.js

I understand it would be a breaking change, but we should at least consider it - if somebody would still need them, AvatarLabel can be replaced with a Label using children callback.

Reply:
Agreed, removed in the last commit